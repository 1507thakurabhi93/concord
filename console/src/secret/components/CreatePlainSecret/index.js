import React from 'react';
import { Field, reduxForm } from 'redux-form';

import { Form, Card, Button, Segment } from 'semantic-ui-react';

const CreatePlainSecret = props => {
  const { handleSubmit, pristine, reset, submitting } = props;
  return (
    <Segment> 
      <h2>Create Plain Secret</h2>
      <form onSubmit={ handleSubmit }>
        <div>
          <label>Concord ID</label>
            <Field
              name="ConcordId"
              component={Form.Input}
              type="text"
              placeholder="Concord ID"
            />
        </div>
        <div>
          <label>Secret Phrase</label>
            <Field
              name="secret"
              component={Form.Input}
              type="text"
              placeholder="Plain Secret Text"
            />
        </div>
        {/* <div>
          <label>Generate Password</label>
            <Field
              name="generatePassword"
              component={Form.Checkbox}
              parse={ () }
            />
        </div>
        <div>
          <label>Store password</label>
          <Field
            name="storePassword"
            component={Form.Input}
            type="password"
            placeholder="Store Password"
          />
        </div>
         */}
        <div>
          <Button type="submit" disabled={pristine || submitting}>Submit</Button>
        </div>

      </form>
    </Segment>
  );
};

export default reduxForm({
  form: 'CreatePlainSecret',
})(CreatePlainSecret);
