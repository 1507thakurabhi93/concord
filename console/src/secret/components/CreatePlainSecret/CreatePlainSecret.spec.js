import React from 'react';
import { shallow, mount } from 'enzyme';
import CreatePlainSecret from './';

import { Form } from 'semantic-ui-react';

describe('CreatePlainSecret Component', () => {

    it('should render with no errors', () => {
        shallow(<CreatePlainSecret />);
    });

    it('should render a text input for the Secret Name', () => {
        const wrapper = shallow(<CreatePlainSecret />);
        expect( wrapper ).toMatchSnapshot();
    });

});