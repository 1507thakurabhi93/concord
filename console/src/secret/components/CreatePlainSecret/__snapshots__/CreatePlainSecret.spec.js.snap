// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePlainSecret Component should render a text input for the Secret Name 1`] = `
<Connect(Form(CreatePlainSecret))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="CreatePlainSecret"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;
